gno.land/pkg/sdk/vm/gas_test.go

```diff
@@ -43,7 +43,7 @@ func TestAddPkgDeliverTxInsuffGas(t *testing.T) {
 			assert.True(t, abort)
 			assert.False(t, res.IsOK())
 			gasCheck := gctx.GasMeter().GasConsumed()
-			assert.Equal(t, int64(3231), gasCheck)
+			assert.Equal(t, int64(3462), gasCheck)
```

Comment:
this is because all of the storage changes, right?

Reply:
I got stuck on this comment for a while because I was ready to scoff it off saying "yes, the store access changed and so did the gas"; but as I started trying to justify exactly _why_ these gas values had changed I realised they had no good reason to change (the number of read/writes on the underlying stores should stay the same with this bug; except for "incorrect" cases like the one of the txtar) and there was probably something funky going on.

And indeed, [there was](https://github.com/gnolang/gno/pull/2319/commits/2b2c7da1b19646fb98388998e2c37a928d073f7f).

In tests, the TransactionStore being set up was only created once, during the setup; and that was always reused subsequently. It was using the root baseStore/iavlStore; thus access to the baseStore/iavlStore did not increase gas.

But even during normal execution, we were getting the stores directly from the MultiStore; rather than going through `sdk.Context.Store`; which returns the stores wrapped in a gas store. IOW, gas usage changed because we were not counting gas when the Gno store touched the DB. Crisis averted.

Thank you!